source/geometries/Cigar.cc

```diff
@@ -0,0 +1,512 @@
+// ----------------------------------------------------------------------------
+// nexus | Cigar.cc
+//
+// Box containing optical fibers
+//
+// The NEXT Collaboration
+// ----------------------------------------------------------------------------
+
+#include "Cigar.h"
+
+#include "FactoryBase.h"
+#include "OpticalMaterialProperties.h"
+#include "Visibilities.h"
+#include "GenericSquarePhotosensor.h"
+
+#include <G4GenericMessenger.hh>
+#include <G4OpticalSurface.hh>
+#include <G4LogicalSkinSurface.hh>
+#include <G4LogicalBorderSurface.hh>
+#include <G4LogicalVolume.hh>
+#include <G4PVPlacement.hh>
+#include <G4NistManager.hh>
+#include <G4Box.hh>
+#include <G4Tubs.hh>
+#include <G4VisAttributes.hh>
+#include <G4SubtractionSolid.hh>
+
+
+using namespace nexus;
+
+REGISTER_CLASS(Cigar, GeometryBase)
+
+namespace nexus {
+
+  Cigar::Cigar():
+    GeometryBase(),
+    cigar_length_ (400 * mm),
+    cigar_width_ (30 * mm),
+    fiber_diameter_(1 * mm),
+    gas_("Ar"),
+    pressure_(0. * bar),
+    coating_ ("TPB"),
+    fiber_type_ ("Y11"),
+    coated_(true)
+  {
+    msg_ = new G4GenericMessenger(this, "/Geometry/Cigar/",
+      "Control commands of geometry Cigar.");
+
+    G4GenericMessenger::Command&  width_cmd =
+      msg_->DeclareProperty("cigar_width", cigar_width_,
+                            "Barrel fiber radius");
+    width_cmd.SetUnitCategory("Length");
+
+    G4GenericMessenger::Command&  length_cmd =
+      msg_->DeclareProperty("cigar_length", cigar_length_,
+                            "Barrel fiber length");
+    length_cmd.SetUnitCategory("Length");
+
+    G4GenericMessenger::Command&  fiber_diameter_cmd =
+      msg_->DeclareProperty("fiber_diameter", fiber_diameter_,
+                            "Fiber diameter");
+    fiber_diameter_cmd.SetUnitCategory("Length");
+
+    msg_->DeclareProperty("gas", gas_, "Gas (Ar or Xe)");
+
+    G4GenericMessenger::Command&  pressure_cmd =
+      msg_->DeclareProperty("pressure", pressure_,
+                            "Vessel pressure");
+    pressure_cmd.SetUnitCategory("Pressure");
+
+    msg_->DeclareProperty("coating", coating_, "Fiber coating (TPB or PTH)");
+    msg_->DeclareProperty("fiber_type", fiber_type_, "Fiber type (Y11 or B2)");
+    msg_->DeclareProperty("coated", coated_, "Coat fibers with WLS coating");
+
+  }
+
+
+
+  Cigar::~Cigar()
+  {
+    delete msg_;
+  }
+
+
+
+  void Cigar::Construct()
+  {
+
+    G4cout << "[Cigar] *** Cigar geometry ***" << G4endl;
```

Comment:
We don't want the code to print out text by default. You should define a configuration parameter to choose if printing verbosity or not and set its value to `False` by default.